### DISCUSSION TITLE
Add support for Go's `go.mod` tool updates

### DISCUSSION THREAD

**Discussion by jamietanna** [renovatebot/renovate#24414] [Suggest an Idea]
### Tell us more.

As part of https://github.com/golang/go/issues/48429 - still in proposal-and-not-yet-approved-or-implemented stage - there will be support in the `go.mod` to manage tools. It'd be great to add support for them when it lands.

We can likely re-use a good chunk of the existing Renovate code for this.

**Comment by secustor** [comment]
Looking at [the proposal](https://go-review.googlesource.com/c/proposal/+/495555/7/design/48429-go-tool-modules.md#85) it seems like it simply adds a tool block, tough it still uses `require` for versions. 
So it should work out of the box 🤔

**Comment by jamietanna** [comment]
We ended up needing a little more for this: https://github.com/renovatebot/renovate/pull/35078
